felix/nftables/ipsets.go

```diff
@@ -413,23 +413,26 @@ func (s *IPSets) tryResync() error {
 	// avoid spawning too many goroutines if there are a large number of sets.
 	routineLimit := make(chan struct{}, 100)
 	defer close(routineLimit)
-	for _, setName := range sets {
-		// Wait for room in the limiting channel.
-		routineLimit <- struct{}{}
-
-		// Start a goroutine to read this set.
-		go func(name string) {
-			// Make sure to indicate that we're done by removing ourselves from the limiter channel.
-			defer func() { <-routineLimit }()
-
-			elems, err := s.nft.ListElements(ctx, "set", name)
-			if err != nil {
-				setsChan <- setData{setName: name, err: err}
-				return
-			}
-			setsChan <- setData{setName: name, elems: elems}
-		}(setName)
-	}
+
+	go func() {
```

Comment:
This works; if I was writing it from scratch, I'd probably use the `errgroup` package now that's in the std lib.

Reply:
Neat, I had missed this one. Updated to use errgroup now.